test/e2e/run-openrpc-api-test-coverage.ts

```diff
@@ -0,0 +1,418 @@
+import testCoverage from '@open-rpc/test-coverage';
+import { parseOpenRPCDocument } from '@open-rpc/schema-utils-js';
+import HtmlReporter from '@open-rpc/test-coverage/build/reporters/html-reporter';
+import ConsoleStreamingReporter from '@open-rpc/test-coverage/build/reporters/console-streaming';
+import ExamplesRule from '@open-rpc/test-coverage/build/rules/examples-rule';
+import JsonSchemaFakerRule from '@open-rpc/test-coverage/build/rules/json-schema-faker-rule';
+
+import {
+  ExampleObject,
+  ExamplePairingObject,
+  MethodObject,
+} from '@open-rpc/meta-schema';
+import openrpcDocument from '@metamask/api-specs';
+import { Call } from '@open-rpc/test-coverage/build/coverage';
+import { ConfirmationsRejectRule } from './api-specs/ConfirmationRejectionRule';
+
+import { Driver, PAGES } from './webdriver/driver';
+
+import { createDriverTransport } from './api-specs/helpers';
+
+import FixtureBuilder from './fixture-builder';
+import {
+  withFixtures,
+  openDapp,
+  unlockWallet,
+  DAPP_URL,
+  ACCOUNT_1,
+} from './helpers';
+
+// eslint-disable-next-line @typescript-eslint/no-require-imports, @typescript-eslint/no-var-requires
+const mockServer = require('@open-rpc/mock-server/build/index').default;
+
+async function main() {
+  const port = 8545;
+  const chainId = 1337;
+  await withFixtures(
+    {
+      dapp: true,
+      fixtures: new FixtureBuilder().build(),
+      disableGanache: true,
+      title: 'api-specs coverage',
+    },
+    async ({ driver }: { driver: Driver }) => {
+      await unlockWallet(driver);
+
+      // Navigate to extension home screen
+      await driver.navigate(PAGES.HOME);
+
+      // Open Dapp
+      await openDapp(driver, undefined, DAPP_URL);
+
+      const transport = createDriverTransport(driver);
+
+      const transaction =
+        openrpcDocument.components?.schemas?.TransactionInfo?.allOf?.[0];
+
+      if (transaction) {
+        delete transaction.unevaluatedProperties;
+      }
+
+      const chainIdMethod = openrpcDocument.methods.find(
+        (m) => (m as MethodObject).name === 'eth_chainId',
+      );
+      (chainIdMethod as MethodObject).examples = [
+        {
+          name: 'chainIdExample',
+          description: 'Example of a chainId request',
+          params: [],
+          result: {
+            name: 'chainIdResult',
+            value: `0x${chainId.toString(16)}`,
+          },
+        },
+      ];
```

Comment:
[nit] Wonder if we could isolate these example modifications into a separate function in a helpers file as well?